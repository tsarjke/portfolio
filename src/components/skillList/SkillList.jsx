import React from 'react';
import './skillList.scss';

const reqSvgs = require.context('./img', true, /\.svg$/);
const svgs = reqSvgs.keys().map((path) => ({ path, file: reqSvgs(path) }));

const SkillList = () => {
  const items = svgs.map(({ path, file }) => {
    const text = path.replace(/([./\d]|svg)/g, '').toUpperCase();
    return (
      <li className="skill-list__item" key={text}>
        <div className="skill-list__content">
          <div className="skill-list__img">
            <img src={file} alt={text} />
          </div>
          <h4 className="skill-list__label">{text}</h4>
        </div>
      </li>
    );
  });

  return <ul className="skill-list">{items}</ul>;
};

export default SkillList;
