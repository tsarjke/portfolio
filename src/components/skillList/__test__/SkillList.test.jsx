import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import SkillList from '../SkillList';

describe('SkillList', () => {
  test('snapshot list with properly placed imgages', () => {
    render(<SkillList />);

    const list = screen.getByTestId('skill-list');
    expect(list).toMatchSnapshot();
  });
});
