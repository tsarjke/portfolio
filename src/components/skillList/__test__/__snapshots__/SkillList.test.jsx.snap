// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillList snapshot list with properly placed imgages 1`] = `
<ul
  class="skill-list"
  data-testid="skill-list"
>
  <li
    class="skill-list__item"
  >
    <div
      class="skill-list__content"
    >
      <div
        class="skill-list__img"
      >
        <img
          alt="TEST"
          src="./img/test.jpg"
        />
      </div>
      <h4
        class="skill-list__label"
      >
        TEST
      </h4>
    </div>
  </li>
</ul>
`;
